tracer/test/Datadog.Trace.Security.IntegrationTests/AspNetCore5AsmToggle.cs

```diff
@@ -24,30 +28,37 @@ public class AspNetCore5AsmToggle : AspNetBase, IDisposable
         public AspNetCore5AsmToggle(ITestOutputHelper outputHelper)
             : base("AspNetCore5", outputHelper, "/shutdown", testName: nameof(AspNetCore5AsmToggle))
         {
+            SetEnvironmentVariable(ConfigurationKeys.Rcm.PollInterval, "500");
         }
 
         [SkippableTheory]
-        [InlineData(true)]
-        [InlineData(false)]
-        [InlineData(null)]
+        [InlineData(true, ApplyStates.ACKNOWLEDGED)]
+        [InlineData(false, ApplyStates.UNACKNOWLEDGED)]
+        [InlineData(null, ApplyStates.ACKNOWLEDGED)]
         [Trait("RunOnWindows", "True")]
-        public async Task TestSecurityToggling(bool? enableSecurity)
+        public async Task TestSecurityToggling(bool? enableSecurity, uint expectedState)
         {
-            SetEnvironmentVariable(ConfigurationKeys.Rcm.PollInterval, "500");
             var url = "/Health/?[$slice]=value";
             var agent = await RunOnSelfHosted(enableSecurity);
-            var settings = VerifyHelper.GetSpanVerifierSettings(enableSecurity);
-            var testStart = DateTime.UtcNow;
+            var settings = VerifyHelper.GetSpanVerifierSettings(enableSecurity, expectedState);
 
             var spans1 = await SendRequestsAsync(agent, url);
 
             agent.SetupRcm(Output, new[] { ((object)new Features() { Asm = new Asm() { Enabled = false } }, "1") }, "FEATURES");
-            await Task.Delay(1000);
+
+            var request1 = await agent.WaitRcmRequestAndReturnLast();
+            CheckAckState(request1, expectedState, null, "First RCM call");
+            // even the request show the applied state seems extra time is needed before it's active
+            await Task.Delay(1500);
 
             var spans2 = await SendRequestsAsync(agent, url);
 
             agent.SetupRcm(Output, new[] { ((object)new Features() { Asm = new Asm() { Enabled = true } }, "2") }, "FEATURES");
-            await Task.Delay(1000);
+
+            var request2 = await agent.WaitRcmRequestAndReturnLast();
+            CheckAckState(request2, expectedState, null, "Second RCM call");
+            // even the request show the applied state seems extra time is needed before it's active
+            await Task.Delay(1500);
```

Comment:
This seems like it might cause flakiness in CI (e.g. if the build agent is being slow for whatever reason, 1.5 seconds might not be enough). Perhaps we should replace this with a loop that waits (until a certain time out?)

Reply:
I previously we were waiting before the requesting the json from the http request. I changed `agent.GetRcmRequest` to `agent.WaitRcmRequestAndReturnLast` to wait for an RcmRequest with client state filled in. To my surprise this is not enough for the system to be ready, and you need to wait more time for the changes to be fully applied. As this seems to all happen asynchronously, I'm not sure what we could wait on. I agree it would better if there was some sort of event that signaled the changes had been fully applied.

Reply:
Well, there are a lot of disadvantages to what I'm about to suggest, but I would argue:
1. When ASM is dynamically turned on or off, we should be sure to write an `Information` level message in the logs
2. If we don't have a better option, using `LogEntryWatcher.WaitForLogEntry` is preferable, as it would allow us to avoid using `Thread.Sleep` or `Task.Delay` in testing, which is asking for trouble ;)

Reply:
I have another PR in this area queued up, so I'll try adding that to PR: https://github.com/DataDog/dd-trace-dotnet/pull/3186